### ISSUE TITLE
Create get_diff and related methods for GeoGraphTimeline

### ISSUE THREAD

**Issue by rdnfn** [ai4er-cdt/gtc-biodiversity#26]
**Description:** for the UI it would be great to have a `GeoGraphTimeline` class method like

```python
    def get_diff(
        self, start_date: datetime.date(), end_data: datetime.date()
    ) -> Tuple[gpd.GeoDataFrame(), gpd.GeoDataFrame()]:
        """Get the node diff and polygon diff between two dates.

        This method returns the node diff and the polygon diff in the GeoGraphTimeline
        between `start_date` and `end_date`.

        The node diff is returned as a gpd.GeoDataFrame with columns 'id' and 'change',
        where 'id' is the unique node id and 'change' is one of
        ['added','removed','unchanged']. Change here indicates whether a node has been
        added, removed or remained unchanged between the dates. Further there could 
        be a column 'rate_of_change' that indicates for each node how much it has
        changed in percentages with a range [-1,1].

        The polygon diff is returned as a gpd.GeoDataFrame with columns 'id', 'geometry'
        and 'change'. The 'geometry' column contains polygons corresponding to changes
        between the times for node with id as in column 'id'. Column 'change' is one of
        ['added','removed','unchanged']. Each node may have multiple entries for
        different parts that have changed.

        Sidenote: If only years are considered, the 1.1.YEAR should be used for the
        dates within the GeoGraphTimeline class.



        Raises:
            NotImplementedError: [description]
        """
```

Additionally, it would be great if the `GeoGraphTimeline` class could have something like a `dates` attribute that contains a list of all the `datetime.date()`´s for the different `GeoGraph`´s in the timeline.

@Croydon-Brixton: does this sound realistic to you, do you think this is implementable? Would you change something about the interface?

**Comment by Croydon-Brixton** [comment]
Yes I think that is implementable and aligns well with the node identification. 

Just a question on this part:
>  Each node may have multiple entries for different parts that have changed.

Do you have an example case in mind?

**Comment by rdnfn** [comment]
Great! Here are two examples of what I was thinking there

1. A node at one side adds area, whilst on the other side it loses area. The loss and addition should be represented as two rows (mulitple rows is what I mean by multiple entries) one with 'added' and the other with 'removed' in the 'change' column.
2. A node looses area on two sides and the removed areas form disconnected polygons. I was thinking it might be easier to add these as two row entries instead of one, but there either should work as long as the geometry is well represented.

**Comment by herbiebradley** [comment]
Closed by #48
